client/ayon_core/tools/publisher/models/create.py

```diff
@@ -0,0 +1,758 @@
+            instances_by_id[instance_id]
+            for instance_id in instance_ids
+        ]
+        try:
+            self._create_context.remove_instances(instances)
+        except CreatorsOperationFailed as exc:
+            self._emit_event(
+                "instances.remove.failed",
+                {
+                    "title": "Instance removement failed",
+                    "failed_info": exc.failed_info
+                }
+            )
+
+    def _on_create_instance_change(self):
+        self._emit_event("instances.refresh.finished")
+
+    def _collect_creator_items(self) -> Dict[str, CreatorItem]:
+        # TODO add crashed initialization of create plugins to report
+        output = {}
+        allowed_creator_pattern = self._get_allowed_creators_pattern()
+        for identifier, creator in self._create_context.creators.items():
+            try:
+                if self._is_label_allowed(
+                    creator.label, allowed_creator_pattern
+                ):
+                    output[identifier] = CreatorItem.from_creator(creator)
+                    continue
+                self.log.debug(f"{creator.label} not allowed for context")
+            except Exception:
+                self.log.error(
+                    "Failed to create creator item for '%s'",
+                    identifier,
+                    exc_info=True
+                )
+
+        return output
+
+    def _get_allowed_creators_pattern(self) -> Union[Pattern, None]:
+        """Provide regex pattern for configured creator labels in this context
+
+        If no profile matches current context, it shows all creators.
+        Support usage of regular expressions for configured values.
+        Returns:
+            (re.Pattern)[optional]: None or regex compiled patterns
+                into single one ('Render|Image.*')
+        """
+
+        task_type = self._create_context.get_current_task_type()
+        project_settings = self._get_current_project_settings()
+
+        filter_creator_profiles = (
+            project_settings
+            ["core"]
+            ["tools"]
+            ["creator"]
+            ["filter_creator_profiles"]
+        )
+        filtering_criteria = {
+            "task_names": self.get_current_task_name(),
+            "task_types": task_type,
+            "host_names": self._create_context.host_name
+        }
+        profile = filter_profiles(
+            filter_creator_profiles,
+            filtering_criteria,
+            logger=self.log
+        )
+
+        allowed_creator_pattern = None
+        if profile:
+            allowed_creator_labels = {
+                label
+                for label in profile["creator_labels"]
+                if label
+            }
+            self.log.debug(f"Only allowed `{allowed_creator_labels}` creators")
+            allowed_creator_pattern = (
+                re.compile("|".join(allowed_creator_labels)))
+        return allowed_creator_pattern
```

Comment:
Shouldn't this be in a controller as opposed to the model?

Reply:
It should not be controller, but hard to tell if it should be done the way it is done now. Like stated in description, this is not "everything is ideal", but "first rough refactor".

Reply:
Cool - not saying it has to be perfect. Just checking what the design goals are so I can review accordingly.